### DISCUSSION TITLE
Certificate Management API wsdl?

### DISCUSSION THREAD

**Discussion by kc7bfi** [AxisCommunications/discussions#127] [VAPIX]
Is there a WSDL for the VAPIX Certificate Management API?

**Comment by mattias-kindborg-at-work** [comment]
In the context of VAPIX or ONVIF?

**Comment by kc7bfi** [comment]
VAPIX

**Reply by mattias-kindborg-at-work** [reply]
I'll see what I can find.

**Comment by mattias-kindborg-at-work** [comment]
Hi @kc7bfi, sorry for the late reply.

If you have access to an Axis device, you can download the three WSDL files directly:

- http://192.168.0.90/wsdl/onvif/devicemgmt.wsdl
- http://192.168.0.90/wsdl/vapix/CertificateService.wsdl
- http://192.168.0.90/wsdl/vapix/HttpConfigurationService.wsdl

The links assume your Axis device is found on IP address 192.168.0.90.

**Reply by kc7bfi** [reply]
Sorry for my late reply, I was on vacation. Thanks so much for these WSDL locations. David

**Reply by kc7bfi** [reply]
Thansk. One more question. Where do I get cert.xsd?

<xs:import namespace="http://www.axis.com/vapix/ws/cert" schemaLocation="http://www.axis.com/vapix/ws/cert/cert.xsd"/>

The URL http://www.axis.com/vapix/ws/cert/cert.xsd returns Error 404

**Reply by Val9000** [reply]
> Hi @kc7bfi, sorry for the late reply.
> 
> If you have access to an Axis device, you can download the three WSDL files directly:
> 
> * http://192.168.0.90/wsdl/onvif/devicemgmt.wsdl
> * http://192.168.0.90/wsdl/vapix/CertificateService.wsdl
> * http://192.168.0.90/wsdl/vapix/HttpConfigurationService.wsdl
> 
> The links assume your Axis device is found on IP address 192.168.0.90.


So if I want to set the TLS mode, client certificate and so on, I've to use the endpoints from [HttpConfigurationService.wsdl] ?

for example, I tried it the following way. 

```xml

<?xml version="1.0" encoding="utf-8"?>
<soap12:Envelope xmlns:soap12="http://www.w3.org/2003/05/soap-envelope">
  <soap12:Body>
    <SetWebServerTlsConfiguration xmlns="http://www.axis.com/vapix/ws/webserver">
      <Configuration>
        <Tls>true</Tls>
        <ConnectionPolicies>
          <Admin>HttpAndHttps</Admin>
        </ConnectionPolicies>
            <CertificateSet>
                <Certificates>
                    <Id>ExampleCertificateID</Id>
                </Certificates>
                <CACertificates></CACertificates>
                <TrustedCertificates></TrustedCertificates>
            </CertificateSet>
        </Configuration>
    </SetWebServerTlsConfiguration>
  </soap12:Body>
</soap12:Envelope>

```

That didn't work out, can you maybe give an valid example ?

**Reply by kc7bfi** [reply]
Yes, I was able to download the WSDLs. However, in the CertificateService.wsdl, it imports an XML schema.

 <xs:import namespace="http://www.axis.com/vapix/ws/cert" schemaLocation="http://www.axis.com/vapix/ws/cert/cert.xsd"/>

I was wondering where I could get a copy of that schema. I tried the location http://www.axis.com/vapix/ws/cert/cert.xsd but it did not return anything. David
